db/common/src/main/java/io/helidon/db/common/AbstractDbExecute.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedInsert(String statementName) {
+        return createNamedInsert(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createInsert(String statement) {
+        return createNamedInsert(generateName(DbStatementType.INSERT, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedUpdate(String statementName) {
+        return createNamedUpdate(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createUpdate(String statement) {
+        return createNamedUpdate(generateName(DbStatementType.UPDATE, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedDelete(String statementName) {
+        return createNamedDelete(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createDelete(String statement) {
+        return createNamedDelete(generateName(DbStatementType.DELETE, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createNamedDmlStatement(String statementName) {
+        return createNamedDmlStatement(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, CompletionStage<Long>> createDmlStatement(String statement) {
+        return createNamedDmlStatement(generateName(DbStatementType.DML, statement), statement);
+    }
+
+    @Override
+    public DbStatement<?, DbResult> createNamedStatement(String statementName) {
+        return createNamedStatement(statementName, statementText(statementName));
+    }
+
+    @Override
+    public DbStatement<?, DbResult> createStatement(String statement) {
+        return createNamedStatement(generateName(DbStatementType.UNKNOWN, statement), statement);
+    }
+
+    /**
+     * Generate a name for a statement.
+     * The default implementation uses {@code SHA-256} so the same name is always
+     * returned for the same statement.
+     * <p>
+     * As there is always a small risk of duplicity, named statements are recommended!
+     *
+     * @param type      type of the statement
+     * @param statement statement that it going to be executed
+     * @return name of the statement
+     */
+    protected String generateName(DbStatementType type, String statement) {
+        String sha256;
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            digest.update(statement.getBytes(StandardCharsets.UTF_8));
+            sha256 = Base64.getEncoder().encodeToString(digest.digest());
+        } catch (NoSuchAlgorithmException ignored) {
+            return "sha256failed";
+        }
+        return type.prefix() + '_' + sha256;
```

Comment:
why type prefix concatenated with SHA-256? Statement hash shall be unique enough. The only exception I see here is common DML vs specific DML type.